infrastructure/instances.tf

```diff
@@ -243,98 +243,296 @@ data "template_file" "nomad_client_script_smasher_smusher" {
+#     # https://github.com/hashicorp/terraform/issues/15978
+
+#     name_prefix = "auto-client-"
+#     image_id = "${data.aws_ami.ubuntu.id}"
+#     instance_type = "${var.client_instance_type}"
+#     security_groups = ["${aws_security_group.data_refinery_worker.id}"]
+#     iam_instance_profile = "${aws_iam_instance_profile.data_refinery_instance_profile.name}"
+#     depends_on = [
+#               "aws_internet_gateway.data_refinery",
+#               "aws_instance.nomad_server_1",
+#               "aws_ebs_volume.data_refinery_ebs",
+#               "aws_instance.pg_bouncer"
+#     ]
+#     user_data = "${data.template_file.nomad_client_script_smusher.rendered}"
+#     key_name = "${aws_key_pair.data_refinery.key_name}"
+#     spot_price = "${var.spot_price}"
+
+#     lifecycle {
+#         create_before_destroy = true
+#     }
+
+#     # I think these are the defaults provided in terraform examples.
+#     # They should be removed or revisited.
+#     root_block_device = {
+#       volume_type = "gp2"
+#       volume_size = 100
+#     }
+# }
+
+# resource "aws_autoscaling_group" "clients" {
+#     name = "asg-clients-${var.user}-${var.stage}"
+#     max_size = "${var.max_clients}"
+#     min_size = "0"
+#     desired_capacity = "2"
+#     health_check_grace_period = 300
+#     health_check_type = "EC2"
+
+#     # 300 seconds so we don't start more than an instance worth of
+#     # downloader jobs at once as we scale up.
+#     default_cooldown = 300
+
+#     # Super important flag. Makes it so that terraform doesn't fail
+#     # every time because it can't acquire spot instances fast enough
+#     # for the autoscaling group.
+#     wait_for_capacity_timeout = "0"
+#     force_delete = true
+#     launch_configuration = "${aws_launch_configuration.auto_client_configuration.name}"
+#     vpc_zone_identifier = ["${aws_subnet.data_refinery_1a.id}"]
+
+#     tag {
+#         key = "Name"
+#         value = "Nomad Client Instance ${var.user}-${var.stage}"
+#         propagate_at_launch = true
+#     }
+
+#     tag {
+#         key = "User"
+#         value = "${var.user}"
+#         propagate_at_launch = true
+#     }
+
+#     tag {
+#         key = "Stage"
+#         value = "${var.stage}"
+#         propagate_at_launch = true
+#     }
+
+# }
+
+# resource "aws_autoscaling_policy" "clients_scale_up" {
+#     name = "asg-clients-scale-up-${var.user}-${var.stage}"
+#     scaling_adjustment = 1
+#     adjustment_type = "ChangeInCapacity"
+#     cooldown = 60
+#     autoscaling_group_name = "${aws_autoscaling_group.clients.name}"
+#     depends_on = ["aws_instance.nomad_server_1"]
+# }
+
+# resource "aws_autoscaling_policy" "clients_scale_down" {
+#     name = "asg-clients-scale-down-${var.user}-${var.stage}"
+#     scaling_adjustment = -1
+#     adjustment_type = "ChangeInCapacity"
+#     cooldown = 60
+#     autoscaling_group_name = "${aws_autoscaling_group.clients.name}"
+#     depends_on = ["aws_instance.nomad_server_1"]
+# }
+
```

Comment:
Can we clean up this dead code? If we ever need to go back to it it'll be in the git history.